index.html

```diff
@@ -604,6 +616,166 @@ <h3>Set the session description</h3>
       </ol>
     </section>
   </section>
+  <section id="rtcicetransport">
+    <h3>
+      {{RTCIceTransport}} extensions
+    </h3>
+    <p>
+      The {{RTCIceTransport}} interface is defined in [[WEBRTC]]. This document extends that interface to allow an
+      application to observe and affect certain actions that an <dfn>ICE agent</dfn> [[RFC5245]] performs.
+    </p>
+    <p>
+      When the [= ICE agent =] has selected a candidate pair to prune, the [= user agent =] MUST [= queue a task =] to <dfn id="rtcicetransport-prune">prune a candidate pair</dfn>:
+    </p>
+    <ol class="algorithm">
+      <li>
+        <p>
+          Let |connection:RTCPeerConnection| be the {{RTCPeerConnection}} object associated with this [= ICE agent =].
+        </p>
+      </li>
+      <li>
+        <p>
+          If <var>connection</var>.{{RTCPeerConnection/[[IsClosed]]}} is
+          <code>true</code>, abort these steps.
+        </p>
+      </li>
+      <li>
+        <p>
+          Let |transport:RTCIceTransport| be the {{RTCIceTransport}} object associated with this candidate pair.
+        </p>
+      </li>
+      <li>
+        <p>
+          Let |candidatePair:RTCIceCandidatePair| be the candidate pair which is being pruned.
+        </p>
+      </li>
+      <li>
+        <p>
+          Let |accepted:boolean| be the result of [= fire an event | firing an event =] named
+          {{RTCIceTransport/icecandidatepairprune}} at |transport|, using {{RTCIceCandidatePairEvent}}, with the
+          {{Event/cancelable}} attribute initialized to <code>true</code>, and the {{RTCIceCandidatePairEvent/local}} and {{RTCIceCandidatePairEvent/remote}} attributes initialized to the local and remote candidates, respectively, of |candidatePair|.
+        </p>
+      </li>
+      <li>
+        <p>
+          If |accepted| is <code>false</code>, instruct the [= ICE agent =] to not prune the candidate pair indicated by |candidatePair|, and instead continue to send and respond to ICE connectivity checks on the candidate pair as before.
+        </p>
+      </li>
+      <li>
+        <p>
+          Otherwise, instruct the [= ICE agent =] to prune the candidate pair indicated by |candidatePair|.
+        </p>
+      </li>
+    </ol>
+    <pre class="idl">
+      partial interface RTCIceTransport {
+        attribute EventHandler onicecandidatepair;
```

Comment:
It seems like these should be symmetric, like onicecandidateadd/onicecanddiateremove.  

I also like "remove" more than "prune".

Reply:
`onicecandidate` was what I was trying to imitate. But after some thought I'm coming around to `onicecandidatepairadd`/`onicecandidatepairremove`.

"Prune" has a specific, and rather different, meaning in the ICE spec (https://www.rfc-editor.org/rfc/rfc8445#section-6.1.2.4). To avoid any confusion, it makes sense to use the more neutral term "remove". In which case I prefer the symmetry of add/remove despite the departure from `onicecandidate`.

Now to make the add/remove events actually symmetric, there should be an `onicecandidatepairremove` event for every discarded candidate pair which was previously announced with `onicecandidatepairadd`, whether the discard was a result of failed connectivity checks or nomination of a different (higher-priority) candidate pair.

I previously considered separate events, i.e. `onicecandidatepair[delete|remove]` and `onicecandidatepairprune`, for these two scenarios, but it might make sense to bundle those into this single event. The event is cancelable if the application has a choice over the removal, otherwise it is not cancelable and calling `preventDefault()` has no effect.

The only scenario when a **_cancelable_** `onicecandidatepairremove` is fired is:
- another candidate pair is nominated for this data stream, and this candidate pair is not currently used for transport
  - calling `preventDefault()` retains the candidate pair for future selection, and connectivity checks continue

This is the also the only scenario addressed by the PR in its current form.

In all other scenarios, a **_non-cancelable_** `onicecandidatepairremove` is fired, eg.:
- connectivity checks fail for this candidate pair
- the network interface associated with the active candidate pair goes away
- this candidate pair was retained previously by calling `preventDefault()` on the event, is not in active use for transport, and the associated network interface goes away
- this candidate pair is not in active use for transport, was retained by the local peer, but not retained by the remote peer, the associated remote candidate is freed, and connectivity check subsequently fails
- removeCandidatePairs(...) is called for this candidate pair.